### PR TITLE
complete pagerduty

### PR DESCRIPTION
**Pull request by mbliakharskyi**
## Description

Please include a summary of the changes and issues addressed in this PR. Also, explain the motivation and context of each update. List any dependencies or prerequisites needed for implementing this change.

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/pager-duty/src/app/setup/service.test.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+    // check if the token in the database is updated
+    await expect(
+      db
+        .select({ accessToken: Organisation.accessToken })
+        .from(Organisation)
+        .where(eq(Organisation.id, organisation.id))
+    ).resolves.toMatchObject([
+      {
+        accessToken: encryptToken,
```
[the remark this answers is not in the document]
Comment (mbliakharskyi) [reply apps/pager-duty/src/app/setup/service.test.ts]: Okay.

**Comment by elbalexandre** [comment]
Tested and works fine, just a missing test for `deleteUser`.

**Comment by mbliakharskyi** [comment]
> Tested and works fine, just a missing test for `deleteUser`.

already did it.

File: apps/pager-duty/src/connectors/users.test.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    });
+
+    test('should throw PagerdutyError when token is invalid', async () => {
+      try {
+        await deleteUsers({ token: 'invalidToken', userId });
+      } catch (error) {
+        expect((error as PagerdutyError).message).toEqual(
+          `Could not delete user with id: ${userId}`
+        );
+      }
```
Comment (elbalexandre) [inline apps/pager-duty/src/connectors/users.test.ts:106]:
```suggestion
        await expect(deleteUsers({ token: 'invalidToken', userId })).rejects.toBeInstanceOf(PagerdutyError);
```

File: apps/pager-duty/src/connectors/users.test.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          }
+          return new Response(undefined, { status: 200 });
+        })
+      );
+    });
+
+    test('should delete user successfully when token is valid', async () => {
+      await expect(deleteUsers({ token: validToken, userId })).resolves.not.toThrow();
+    });
+
```
Comment (elbalexandre) [inline apps/pager-duty/src/connectors/users.test.ts:98]:
```suggestion
test('should not throw when the user is not found', async () => ...)
```

**Review by elbalexandre — APPROVED** [review]

**Pull request merged**